core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequest.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.resource
+
+/**
+ * An executor resource request. This is used in conjuntion with the ResourceProfile to
+ * programmatically specify the resources needed for an RDD that will be applied at the
+ * stage level.
+ * There are alternative constructors for working with Java.
+ *
+ * @param resourceName Name of the resource
+ * @param amount Amount requesting
+ * @param units Units of amount for things like Memory, default is no units, only byte
```

Comment:
If we only support bytes types is a string the way to do this?

Reply:
I was leaving it a string in case future enhancements required types other then bytes so we wouldn't have to change API, but we can restrict for now and add something else later if you prefer?